### PR TITLE
Currency field updates

### PR DESCRIPTION
**Pull request by chromaticWaster**
## Description:

Updates to the currency field and other components based on testing feedback received.

### Updates to the currency field component. 

* Updates to properties for input formatting and currency symbol.
* Label positioning update in base class.
* Updated keydown event function for currency field updates.
* Styling updates.

(*Link to bug/feature from [Issues](../issues) and how you resolve it*)

### Test Results

image

---

### Delete key presses

**Caret positioned at decimal separator.**

caret delete key decimal separator

**Caret positioned at thousand separator**

caret delete key thousand separator

**Caret positioned at a number**

caret delete key at number

---

### Label position

Label position and transitioning was inconsistent when the component had its value property pre-populated.
 
**Before** 

image

**After**

image

---

### Color field component

**Before**

color field label transition

**After**

color field label transition fix

---

### Decimal precision

Decimal precision set to 4 for this example.

decimal precision

### All Submissions:

* [x] I have followed the guidelines in the Contributing document.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/currency-field/CurrencyField.ts [hunk 1]
```diff
@@ -193,17 +132,38 @@ export class CurrencyField extends OmniFormElement {
...
 
-    _formatToCurrency(preFormattedValue: number): string {
+    // Format the numeric value to a currency formatted string value.
+    _formatToCurrency(preFormattedValue: number | string): string {
+        if (preFormattedValue === 0) {
+            return preFormattedValue.toString();
+        }
+
         if (!preFormattedValue) {
+            console.log('No value provided');
```
Comment (BOTLANNER) [inline src/currency-field/CurrencyField.ts:142]: Do we want console logging in our components like this?

**Review by BOTLANNER — APPROVED** [review]

File: src/currency-field/CurrencyField.ts (hunk 1 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Removing as requested.

File: src/currency-field/CurrencyField.ts [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
-export const CURRENCY_SEPARATOR = {
-    AMOUNT: 1,
-    INVERSE_AMOUNT: 2,
-    CENTS: 3,
-    INVERSE_CENTS: 4
-};
-
 /**
- * A currency input control that formats input based on currency and locale.
+ * A input control to enter a formatted currency value.
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:8]:
```suggestion
 * Control to enter a formatted currency value.
```

File: src/currency-field/CurrencyField.ts [hunk 3]
```diff
@@ -28,8 +20,10 @@ export const CURRENCY_SEPARATOR = {
  *  data="{id: 123, name: 'Debit'}"
  *  hint="Required"
  *  error="Please enter the correct amount"
- *  locale="en-US"
- *  currency="USD"
+ *  currency-symbol="$"
+ *  thousands-separator=","
+ *  decimalSeparator="."
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:25]:
```suggestion
 *  fractional-separetor="."
```

File: src/currency-field/CurrencyField.ts [hunk 4]
```diff
@@ -28,8 +20,10 @@ export const CURRENCY_SEPARATOR = {
...
+ *  decimal-precision=2
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:26]:
```suggestion
 *  fractional-precision=2
```

File: src/currency-field/CurrencyField.ts [hunk 5]
```diff
@@ -57,31 +51,38 @@ export class CurrencyField extends OmniFormElement {
     private _inputElement: HTMLInputElement;
 
     /**
-     * The currency for the input.
-     * @attr
+     * The currency symbol.
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:54]:
```suggestion
     * Currency symbol.
```

File: src/currency-field/CurrencyField.ts [hunk 6]
```diff
@@ -269,20 +228,75 @@ export class CurrencyField extends OmniFormElement {
...
+            this.value = this._formatToFloat(this._stringValue);
+
+            await this.updateComplete;
+            // Set caret position after value is formatted.
+            if (input.value.length === this._stringValue.length) {
+                this._inputElement.setSelectionRange(caretPosition, caretPosition);
+            }
+        }
+
+        // If hitting backspace with the carat in the position of the first decimal then remove the entire cent value.
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:260]:
```suggestion
        // If hitting backspace with the caret in the position of the first decimal then remove the entire cent value.
```

File: src/currency-field/CurrencyField.ts [hunk 7]
```diff
@@ -269,20 +228,75 @@ export class CurrencyField extends OmniFormElement {
...
+        if (input.value.charAt(caretPosition - 1) === this.decimalSeparator && e.key.toLowerCase() === 'backspace') {
+            this._stringValue = input.value.substring(0, input.value.indexOf(this.decimalSeparator));
             e.preventDefault();
             // Set value prop to float value
             this.value = this._formatToFloat(this._stringValue);
             return;
         }
 
-        // Copy currency field selection to clipboard if ctrl + c is pressed on the keyboard.
+        // Delete cents if delete button is clicked.
```
Comment (stefan505) [inline src/currency-field/CurrencyField.ts:282]:
```suggestion
        // Delete fractions (cents) if delete button is clicked.
```

**Review by stefan505 — CHANGES_REQUESTED** [review]

File: src/currency-field/CurrencyField.ts (hunk 2 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Updated as requested

File: src/currency-field/CurrencyField.ts (hunk 3 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Updated

File: src/currency-field/CurrencyField.ts (hunk 4 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Updated

File: src/currency-field/CurrencyField.ts (hunk 5 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Updated similar implementation done on other components

File: src/currency-field/CurrencyField.ts (hunk 6 above)
Comment (chromaticWaster) [reply src/currency-field/CurrencyField.ts]: Updated

**Comment by chromaticWaster** [comment]
Latest test results 

image

Note test amount lower due to removal of Numeric field component being removed.

**Review by stefan505 — APPROVED** [review]

**Pull request merged**